<h3 style="margin:0px">Class: org.jclouds.chef.features.OrganizationApiExpectTest (2 methods) </h3><br><pre style="padding:0"><code class="html"><div id="tags"><button id="10"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('10')" data-toggle="tooltip" title="Executes methods or other tests from the same test unit"><kbd id="tag-10"class="label-info"style="display: inline-block;font-size:7pt" >ExecutionTester&nbsp;(2)</kbd></button>&nbsp;<button id="3"style="border-width:0;padding:0;background:#282B2E"  onclick="filter('3')" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution"><kbd id="tag-3"class="label-info"style="display: inline-block;font-size:7pt" >ExceptionVerifier&nbsp;(2)</kbd></button>&nbsp;</div></code></pre><center><button onclick="showBlocks()" style="margin:6px">Update filter <span class="glyphicon glyphicon-refresh" aria-hidden="true"></span> </button></center><br>
<pre class="type-3 type-10 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(expectedExceptions=ResourceNotFoundException.class) public void testDeleteGroupFailsOn404(){
  ChefApi api=requestSendsResponse(signed(HttpRequest.builder().method("DELETE").endpoint("http://localhost:4000/groups/foo").addHeader("X-Chef-Version",ChefApiMetadata.DEFAULT_API_VERSION).addHeader("Accept",MediaType.APPLICATION_JSON).build()),HttpResponse.builder().statusCode(404).build());
  api.organizationApi().get().deleteGroup("foo");
}

</code></pre>

<pre class="type-3 type-10 "><code><span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Verifies that exceptions are thrown during the test case execution">ExceptionVerifier</span>&nbsp;<span class="label label-info" style="display: inline-block;" data-toggle="tooltip" title="Executes methods or other tests from the same test unit">ExecutionTester</span>&nbsp;<span class=" glyphicon glyphicon-comment" aria-hidden="true" label-info" data-toggle="tooltip" title="This method/test case: 
- Verifies that exceptions are thrown during the test case execution
- Executes methods or other tests from the same test unit
"></span><br>
@Test(expectedExceptions=ResourceNotFoundException.class) public void testUpdateGroupFailsOn404(){
  ChefApi api=requestSendsResponse(signed(HttpRequest.builder().method("PUT").endpoint("http://localhost:4000/groups/admins").addHeader("X-Chef-Version",ChefApiMetadata.DEFAULT_API_VERSION).addHeader("Accept",MediaType.APPLICATION_JSON).payload(payloadFromResourceWithContentType("/group-update.json",MediaType.APPLICATION_JSON)).build()),HttpResponse.builder().statusCode(404).build());
  Group group=Group.builder("admins").client("abiquo").group("admins").user("nacx").build();
  api.organizationApi().get().updateGroup(group);
}

</code></pre>

<script>$(document).ready(function() {
  $('pre code').each(function(i, block) {
    hljs.highlightBlock(block);
  });
});</script>
